docs/source/1.0/spec/aws/aws-json-1_0-protocol.rst

```diff
@@ -61,4 +61,208 @@ See
+      - JSON ``string`` value that is base64 encoded.
+    * - ``boolean``
+      - JSON boolean
+    * - ``byte``
+      - JSON number
+    * - ``short``
+      - JSON number
+    * - ``integer``
+      - JSON number
+    * - ``long``
+      - JSON number
+    * - ``float``
+      - JSON number
+    * - ``double``
+      - JSON number
+    * - ``bigDecimal``
+      - JSON number. Unfortunately, this protocol serializes bigDecimal
+        shapes as a normal JSON number. Many JSON parsers will either
+        truncate the value or be unable to parse numbers that exceed the
+        size of a double.
+    * - ``bigInteger``
+      - JSON number. Unfortunately, this protocol serializes bigInteger
+        shapes as a normal JSON number. Many JSON parsers will either
+        truncate the value or be unable to parse numbers that exceed the
+        size of a double.
+    * - ``string``
+      - JSON string
+    * - ``timestamp``
+      - JSON number (default). This protocol uses ``epoch-seconds``, also
+        known as Unix timestamps, in JSON payloads represented as a double.
+    * - ``document``
+      - Any JSON value
+    * - ``list``
+      - JSON array
+    * - ``set``
+      - JSON array. A set is serialized identically as a ``list`` shape,
+        but only contains unique values.
+    * - ``map``
+      - JSON object
+    * - ``structure``
+      - JSON object. Each member value provided for the structure is
+        serialized as a JSON property where the property name is the same
+        as the member name. The :ref:`jsonName-trait` can be used to serialize
+        a property using a custom name. A null value MAY be provided or
+        omitted for a :ref:`boxed <box-trait>` member with no observable
+        difference.
+    * - ``union``
+      - JSON object. A union is serialized identically as a ``structure``
+        shape, but only a single member can be set to a non-null value.
+
+
+~~~~~~~~~~~~~~~~~~~~~~~~
+Empty body serialization
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+When an operation has no modeled input or if no input parameters are provided,
+a client MUST send an empty JSON object (``{}``) as input. Server-side
+implementations MUST also accept empty request bodies.
+
+When an operation has no modeled output or no output parameters are provided,
+server implementations SHOULD send an empty body. The server MUST still send
+the protocol's ``Content-Type`` header in this case. Clients MUST also accept
+an empty JSON object.
+
+
+.. awsJson1_0-errors:
+
+-----------------------------
+Operation error serialization
+-----------------------------
+
+Error responses in the ``awsJson1_0`` protocol are serialized identically to
+standard responses with one additional component to distinguish which error
+is contained. The component MUST be one of the following: an additional header
+with the name ``X-Amzn-Errortype``, a body field with the name ``code``, or a
+body field named ``__type``. New server-side protocol implementations SHOULD
```

Comment:
Can this SHOULD be moved in front of the MUST? Server implementations should see that information first.